src/js/common/schemaform/SaveFormLink.jsx

```diff
@@ -28,35 +28,34 @@ class SaveFormLink extends React.Component {
       saveForm,
       savedStatus
     } = this.props;
-    let content = (<div>
-      <a onClick={this.openLoginModal}>Sign in before saving your application</a>
-      <LoginModal
-          key={1}
-          title="Sign in to save your application"
-          onClose={this.closeLoginModal}
-          visible={this.state.modalOpened}
-          user={this.props.user}
-          onUpdateLoginUrl={this.props.onUpdateLoginUrl}/>
-    </div>);
+    let content = <a onClick={saveForm}>Save and come back later</a>;
 
-    if (this.props.user.login.currentlyLoggedIn) {
-      content = <a onClick={saveForm}>Save and come back later</a>;
+    if (!this.props.user.login.currentlyLoggedIn) {
+      // if we have a noAuth status, that means they tried to save and got a 401, which
+      // likely means their session is expired (since they were logged in before)
+      content = (<div>
+        {savedStatus === SAVE_STATUSES.noAuth
+            ? <span>Sorry, your session has expired. <a onClick={this.openLoginModal}>Please sign in again</a></span>
+            : <a onClick={this.openLoginModal}>Sign in before saving your application</a>}
+        <LoginModal
+            key={1}
+            title="Sign in to save your application"
+            onClose={this.closeLoginModal}
+            visible={this.state.modalOpened}
+            user={this.props.user}
+            onUpdateLoginUrl={this.props.onUpdateLoginUrl}/>
+      </div>);
     }
 
-    if (savedStatus === SAVE_STATUSES.noAuth) {
-      content = <p>no-auth message</p>;
-    } else if (savedStatus === SAVE_STATUSES.failure) {
-      content = <p>failure message</p>;
+    if (savedStatus === SAVE_STATUSES.failure) {
+      content = <span>failure message</span>;
     } else if (savedStatus === SAVE_STATUSES.pending) {
-      content = <p>spinner or something</p>;
+      content = <span>spinner or something</span>;
```

Comment:
The jumpiness of switching between content with and without a surrounding `<p>` tag was annoying, though I realize we'll likely replace these messages.